tests/utils/database.test.ts

```diff
@@ -0,0 +1,184 @@
+          'src/utils/dataTest.json'
+        );
+        expect(returnValue).to.equal(false);
+      });
+    });
+  });
+  context('load', () => {
+    context('when file exists and has data', () => {
+      before(async function () {
+        await writeFileSync(
+          'src/utils/data.json',
+          JSON.stringify({ foo: 'bar' })
+        );
+      });
+      after(async function () {
+        await unlinkSync('src/utils/data.json');
+      });
+      it('expect load to return data', async () => {
+        const returnValue = await database.load();
+        expect(returnValue).to.eql({ foo: 'bar' });
+      });
+    });
+    context('when file exists and has data - using file path over ride', () => {
+      before(async function () {
+        await writeFileSync(
+          'src/utils/dataTest.json',
+          JSON.stringify({ foo: 'bar' })
+        );
+      });
+      after(async function () {
+        await unlinkSync('src/utils/dataTest.json');
+      });
+      it('expect load to return data', async () => {
+        const returnValue = await database.load('src/utils/dataTest.json');
+        expect(returnValue).to.eql({ foo: 'bar' });
+      });
+    });
+    context('when file exists and has no data', () => {
+      before(async function () {
+        await writeFileSync('src/utils/data.json', '');
+      });
+      after(async function () {
+        await unlinkSync('src/utils/data.json');
+      });
+      it('expect load to return error', async () => {
+        const returnValue = await database.load();
+        expect(returnValue).to.have.own.property('error');
+      });
+    });
+    context('when file does not exist', () => {
+      it('expect load to return error', async () => {
+        const returnValue = await database.load();
+        expect(returnValue).to.have.own.property('error');
+      });
+    });
+  });
+  context('save', () => {
+    context('when file exists and is being saved with valid data', () => {
+      before(async function () {
+        await writeFileSync(
+          'src/utils/data.json',
+          JSON.stringify({ foo: 'bar' })
+        );
+      });
+      after(async function () {
+        await unlinkSync('src/utils/data.json');
+      });
+      it('expect save to save data', async () => {
+        await database.save({ foo: 'foo' });
+        const returnValue = await readFileSync('src/utils/data.json', 'utf-8');
+        expect(JSON.parse(returnValue)).to.eql({ foo: 'foo' });
+      });
+    });
+    context(
+      'when file does not exist and is being saved with valid data',
+      () => {
+        after(async function () {
+          await unlinkSync('src/utils/data.json');
+        });
+        it('expect save to save data', async () => {
+          await database.save({ foo: 'foo' });
+          const returnValue = await readFileSync(
+            'src/utils/data.json',
+            'utf-8'
+          );
+          expect(JSON.parse(returnValue)).to.eql({ foo: 'foo' });
```

Comment:
nit: you could also assert on the json string. It's not unreadable.

Reply:
done